### PR TITLE
Simplify embeddings records reading module

### PR DESCRIPTION
**Pull request by cxgoogle**
As specified in the module docstring, I changed the module to be used on embeddings files generated by the CXR Foundation model only.
- The existing mix of rigidity (use of image ids contained in tfrecords) + flexibility (able to specify feature keys) was rather hard to follow. I think it's worth the tradeoff of making things more simple, and then adding flexibility if/when needed.
- I also don't think that this package should/will be responsible for dealing with more general uses cases of creating/manipulating TFRecord datasets. Hence, we should allow it to be more focused.

Therefore I removed:
- Configurable params such as embeddings size and feature keys of the serialized tf.Example data.
- Logic that parses/uses image/id contained in embeddings files. I consider the image id values in the MIMIC embeddings database to be artifacts which should be ignored. Perhaps the generated field is important for use beyond just that dataset. If so, please let me know.
- Filtering rules in `get_dataset` based on weights and label values. The user can always add additional map/filter calls to the returned dataset, for various use-cases.

I'd like some feedback on whether this is going in the right direction. Lots of usage references to update in other modules and the two current notebooks.

It's possible that I've made certain parts less optimized. Please point them out if relevant.

Testing instructions:
- Open Colab for branch: https://colab.research.google.com/github/Google-Health/imaging-research/blob/add-simple-read/cxr-foundation/MIMIC_Embeddings_Demo.ipynb
- Change git clone line to to target branch: `!git clone -b add-simple-read https://github.com/Google-Health/imaging-research.git`

### CODE REVIEW

File: cxr-foundation/cxr_foundation/train_lib.py [hunk 1]
```diff
@@ -1,179 +0,0 @@
...
-    return features, label, weight, image_id
-
-  dataset = dataset.map(_lookup_label).map(_lookup_weight)
-
-  return dataset.map(lambda features, label, weight, image_id:
-                     (features, label, weight)).filter(
-                         filter_negative_labels).filter(filter_zero_weights)
-
-
-def create_model(heads,
```
Comment (cxgoogle) [inline cxr-foundation/cxr_foundation/train_lib.py:122]: I just moved this function to another module. Not intending to remove it.

File: cxr-foundation/cxr_foundation/embeddings_data.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                                default_value=tf.constant(
+                                    0.0, shape=[constants.DEFAULT_EMBEDDINGS_SIZE]))
+  }
+  parsed_tensors = tf.io.parse_example(serialized_example, features=features)
+  return parsed_tensors[constants.EMBEDDING_KEY]
+
+
+def get_dataset(
+    filenames: Iterable[str],
+    labels: Iterable[int]
```
Comment (cxgoogle) [inline cxr-foundation/cxr_foundation/embeddings_data.py:150]: Not sure how this does performance wise, relative to the previous creation of the static hash map from the input labels dict.

File: cxr-foundation/cxr_foundation/embeddings_data.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+  Returns:
+    The Dataset, containing for each element: (embeddings, label)
+  """
+  ds_embeddings = tf.data.TFRecordDataset(filenames, num_parallel_reads=tf.data.AUTOTUNE).map(
+        parse_embedding_from_example
+   )
+  ds_labels = tf.data.Dataset.from_tensor_slices(labels)
+
+  return tf.data.Dataset.zip((ds_embeddings, ds_labels))
```
Comment (cxgoogle) [inline cxr-foundation/cxr_foundation/embeddings_data.py:167]: Ref:
- https://www.tensorflow.org/api_docs/python/tf/data/Dataset#from_tensor_slices
- https://www.tensorflow.org/tutorials/load_data/tfrecord

I was/am actually rather confused about how the model training code doesn't reference label values at all. Then I inferred that they expect the elements of the Datasets to be: (value, label). Is this correct? Is there also an assumption of the weight being a possible third param?

Comment (asellergren) [reply cxr-foundation/cxr_foundation/embeddings_data.py:167]: Yes, the model training code expects a tuple of (value, label, weight).

**Comment by cxgoogle** [comment]
Update: Now RFR.

I ended up only adding new modules and modifying the MIMIC demo notebook. There were too many pieces to update the corresponding library functions too, especially since a few things have changed. I suggest we merge this first, and then update the lib and other notebook.

Summary:

* Add `embeddings_data` module based on `train_lib`. Stripped all logic that parses/uses image/id contained in embeddings files.
* The function that creates a tf Datasets now just require a list of embedding file paths and the labels.
* Add some util functions to new `gcs.py` and `mimic.py`.
* Notebook: Update some package dependency versions
* Notebook: Use new `embeddings_data` module and remove unneeded `train_model` function

Note: The main purpose of train_model is to help create the datasets and kick off model training, but is not needed in this context. It seems more useful when running the command line script with only an input CSV. Will evaluate and maybe refactor in the follow-up PR.

Testing instructions:
* Open Colab for branch: https://colab.research.google.com/github/Google-Health/imaging-research/blob/add-simple-read/cxr-foundation/MIMIC_Embeddings_Demo.ipynb
* Change git clone line to to target branch: !git clone -b add-simple-read https://github.com/Google-Health/imaging-research.git

**Review by asellergren — APPROVED** [review]
